src/api/services/item.service.ts

```diff
@@ -30,16 +61,57 @@ class ItemService {
   }
 
   async getById(id: string) {
-    const result = await this.fastify.db.select().from(itemTable).where(eq(itemTable.id, id))
-    if (!result[0]) throw new NotFoundError(`Item with ID: ${id} does not exist.`)
-    return result[0]
+    try {
+      const [ item ] = await this.fastify.db
+        .select()
+        .from(itemTable)
+        .where(eq(itemTable.id, id))
+        .execute()
+
+      if (!item) {
+        throw new NotFoundError(`Item with ID: ${id} does not exist.`)
+      }
+
+      const categories = await this.fastify.db
+        .select({ categoryId: categoryTable.id })
+        .from(itemToCategoryTable)
+        .where(eq(itemToCategoryTable.itemId, id))
+        .innerJoin(categoryTable, eq(itemToCategoryTable.categoryId, categoryTable.id))
+        .execute()
+
+      return {
+        ...item,
+        categoryId: categories.map((category) => category.categoryId)
+      }
+    } catch (error) {
+      throw new InternalServerError(`Failed to get item with ID: ${id}`, error)
+    }
   }
 
   async patchById(id: string, payload: ItemUpdate) {
     await this.getById(id)
+
+    const { categoryId, ...itemData } = payload
+    const categories = Array.isArray(categoryId) ? categoryId : [categoryId]
+
     try {
-      const result = await this.fastify.db.update(itemTable).set(payload).where(eq(itemTable.id, id)).returning()
-      return result[0]
+      const [ updatedItem ] = await this.fastify.db
+        .update(itemTable)
+        .set(itemData)
+        .where(eq(itemTable.id, id))
+        .returning()
+
+      if (categoryId !== undefined) {
+        await this.fastify.db
+          .delete(itemToCategoryTable)
+          .where(eq(itemToCategoryTable.itemId, id))
+
+        await this.fastify.db
+          .insert(itemToCategoryTable)
+          .values(categories.map((categoryId) => ({ itemId: id, categoryId })))
```

Comment:
you are deleting all categories and then re-inserting them... this is not a good practice.